online-fishing-store/src/app/pages/main-page/products/products.component.ts

```diff
@@ -10,9 +12,11 @@ export class ProductsComponent implements OnInit {
   @Input() public products: Product[];
   @Output() public selectedProduct: EventEmitter <number> = new EventEmitter<number>();
   constructor(
+    private router: Router
   ) {}
 
-  public onSelectProduct(product){
+  public onSelectProduct(product): void {
```

Comment:
You should emit information about select product for parent component. And after parent component should already redirect